pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -335,6 +341,32 @@ func (m *mgr) createProject(name string, cond condition.Cond, obj runtime.Object
 	})
 }
 
+// deleteSystemProject deletes the system project(s) for a cluster in preparation for deleting the cluster namespace.
+// Normally, the webhook prevents deleting the system project, so Rancher needs to use the sudo user to force it.
+// Otherwise, the deleted namespace will be stuck terminating because it cannot garbage collect the project.
+func (m *mgr) deleteSystemProject(obj runtime.Object, controller string) error {
+	metaAccessor, err := meta.Accessor(obj)
+	if err != nil {
+		return err
+	}
+	projects, err := m.projectLister.List(metaAccessor.GetName(), systemProjectLabels.AsSelector())
+	if err != nil {
+		return err
+	}
+	bypassClient, err := m.projects.WithImpersonation(controllers.WebhookImpersonation())
```

Comment:
Nit: it might be good to first create the bypassClient, then list the projects, then loop over them. May result in slightly improved readability and error handling, since if Rancher fails to create the client, it wouldn't have wastefully listed the projects.